Drivers/Makefile

```diff
@@ -0,0 +1,272 @@
+SERIES = $(shell echo $(PROJECT_TARGET) | cut -c6-7)
+LINE = $(shell echo $(PROJECT_TARGET) | cut -c8-9)
+EXTRA = $(shell echo $(PROJECT_TARGET) | cut -c10-)
+
+SERIES_CAP = $(shell echo $(SERIES) | tr '[:lower:]' '[:upper:]')
+EXTRA_CAP = $(shell echo $(EXTRA) | tr '[:lower:]' '[:upper:]')
+SERIES_GENERIC = stm32$(SERIES)xx
+SERIES_GENERIC_CAP = STM32$(SERIES_CAP)xx
+SERIES_LINE = stm32$(SERIES)$(LINE)
+SERIES_LINE_CAP = STM32$(SERIES_CAP)$(LINE)
+SERIES_LINE_GENERIC = $(SERIES_LINE)xx
+SERIES_LINE_GENERIC_CAP = $(SERIES_LINE_CAP)xx
+
+TARGET = $(PROJECT_TARGET)
+
+
+######################################
+# building variables
+######################################
+# debug build?
+DEBUG = 1
+# optimization
+OPT = -Og
+
+
+#######################################
+# paths
+#######################################
+# Build path
+BUILD_DIR = $(PROJECT_BUILD_DIR)
+
+######################################
+# source
+######################################
+# C sources
+C_SOURCES =  \
+$(PROJECT_C_SOURCES) \
+$(filter-out %template.c, $(wildcard $(SERIES_GENERIC)/$(SERIES_GENERIC_CAP)_HAL_Driver/Src/*.c)) \
+$(SERIES_GENERIC)/system_$(SERIES_GENERIC).c \
+$(SERIES_GENERIC)/$(SERIES_GENERIC)_hal_init.c \
+$(SERIES_GENERIC)/$(SERIES_GENERIC)_hal_timebase_tim.c \
+$(wildcard FreeRTOS-Kernel/*.c) \
+FreeRTOS-Kernel/portable/GCC/ARM_CM4F/port.c \
+$(wildcard common/Src/*.c)
+
+# ASM sources
+ASM_SOURCES =  \
+$(SERIES_GENERIC)/$(SERIES_LINE)/startup_$(SERIES_LINE_GENERIC).s
+
+# ASM sources
+ASMM_SOURCES = 
+
+
+#######################################
+# binaries
+#######################################
+PREFIX = arm-none-eabi-
+# The gcc compiler bin path can be either defined in make command via GCC_PATH variable (> make GCC_PATH=xxx)
+# either it can be added to the PATH environment variable.
+ifdef GCC_PATH
+CC = $(GCC_PATH)/$(PREFIX)gcc
+AS = $(GCC_PATH)/$(PREFIX)gcc -x assembler-with-cpp
+CP = $(GCC_PATH)/$(PREFIX)objcopy
+SZ = $(GCC_PATH)/$(PREFIX)size
+else
+CC = $(PREFIX)gcc
+AS = $(PREFIX)gcc -x assembler-with-cpp
+CP = $(PREFIX)objcopy
+SZ = $(PREFIX)size
+endif
+HEX = $(CP) -O ihex
+BIN = $(CP) -O binary -S
+ 
+#######################################
+# CFLAGS
+#######################################
+# cpu
+CPU = -mcpu=cortex-m4
+
+# fpu
+FPU = -mfpu=fpv4-sp-d16
+
+# float-abi
+FLOAT-ABI = -mfloat-abi=hard
+
+# mcu
+MCU = $(CPU) -mthumb $(FPU) $(FLOAT-ABI)
+
+# macros for gcc
+# AS defines
+AS_DEFS = 
+
+# C defines
+C_DEFS =  \
+USE_HAL_DRIVER \
+$(SERIES_LINE_GENERIC_CAP) \
+$(SERIES_GENERIC_CAP)
+
+C_DEFS := $(addprefix -D,$(C_DEFS))
+
+# AS includes
+AS_INCLUDES = 
+
+# C includes
+C_INCLUDES =  \
+$(PROJECT_C_INCLUDES) \
+$(SERIES_GENERIC)/$(SERIES_GENERIC_CAP)_HAL_Driver/Inc \
+$(SERIES_GENERIC)/$(SERIES_GENERIC_CAP)_HAL_Driver/Inc/Legacy \
+$(SERIES_GENERIC)/CMSIS/Device/ST/$(SERIES_GENERIC_CAP)/Include \
+$(SERIES_GENERIC)/CMSIS/Include \
+FreeRTOS-Kernel/include \
+FreeRTOS-Kernel/portable/GCC/ARM_CM4F \
```

Comment:
In what situation do we need to configure FreeRTOS differently for different systems?

Reply:
FreeRTOS probably shouldn't be different for each system (the only thing I can think of is clocking but I have to double check). Otherwise, stuff might break since the BSP understands FreeRTOS should be a specific way

Reply:
makes sense